src/codelabs/async-await/index.md

```diff
@@ -0,0 +1,346 @@
+Dart to run asynchronous code via the `async` and `await` keywords.
+
+In addition to explanations, this codelab includes embedded editors that show 
+you working code examples and coding exercises. You can use these editors to 
+test your knowledge by completing the exercises.
+
+Before you begin this codelab, you should:
+
+* Have programming experience in another language
+* Know basic syntax in Dart 
+* Have some experience writing asynchronous code in another language
+
+When you finish this codelab, you'll know the following: 
+
+* How to use the `async` keyword
+* How to use the `await` keyword
+* When to use `async` and `await` in your code
+* How your code executes when using the `async/await` keywords
+* How to handle errors from an asynchronous call using `try/catch` expressions 
+in async functions.
+* Where to find more information for the `async/await` material 
+covered in this codelab.
+
+
+## Why async matters
+
+Asynchronous operations let your program complete other work while waiting for 
+an operation to finish. Some common processes that require asynchronous 
+operations include: 
+
+* Fetching data over a network
+* Writing to a database
+* Reading data from a file
+
+To implement these features, you will need to learn the Dart syntax for
+futures, async, and await. In Dart you cannot use synchronous code to handle
+asynchronous operations.
+
+Consider the following example that fails to print the desired user order to the
+console: 
+
+
+<iframe style="margin-top: 25px; margin-bottom: 25px" frameborder="no" height="420" src="https://dartpad.dartlang.org/experimental/embed-new.html?id=5c8c7716b6b4284842f15fe079f61e47" width="100%"></iframe>
+
+
+In the above code:
+
+* `getUserOrder` is an asynchronous function that waits four seconds before
+returning a string describing the user's order: a large latte. (For a further
+explanation you can [find out more about
+Future.delayed]({{site.dart_api}}/Future.delayed.html))
+* `createOrderMessage` calls `getUserOrder` as if it were a synchronous function
+ -- it invokes `getUserOrder` and immediately prints the result. 
+* Since `getUserOrder` waits four seconds before returning the user's order,
+`createOrderMessage` never actually prints the user's order. Instead the console
+prints the value immediately returned by `createOrderMessage` which
+turns out to be `Instance of '_Future<String>'`.
+
+What is a future, and how do you write code to handle the asynchronous function
+ `getUserOrder`?
+ 
+ {{ site.alert.secondary }} 
+ Quick review: 
+ * Asynchronous operations let a program do work while waiting for an operation to finish.
+ * In Dart, you can use Futures, `async` and `await` to write declarative asynchronous code.
+ {{ site.alert.end }}
+
+
+## Async in Dart: What is a future?
+
+In Dart, a
+[Future]({{site.dart_api}}/Future-class.html)
+represents the result of an asynchronous operation. A future object has
+two states: uncompleted or completed.
+
+{{ site.alert.note }}
```

Comment:
'Uncompleted' is a ~~special term in~~ **Dart term** referring to the state of a future 
 before it has produced a result.
-i think putting information in a prep phrase sometimes can slow readers down; like one of my earlier comments, I think using Dart as an adjective is more clear because, like **syntax**, **term** is the noun, and **Dart** is used to describe it.

Reply:
Yeah, there may be better ways to do this. What I'm going for here is this:

I want to introduce readers to the fact that `uncompleted` is a legitimate descriptor in Dart, and not a typo.